File: library/src/main/java/com/zsp/library/guide/materialintroview/animation/AnimationListener.java
package com.zsp.library.guide.materialintroview.animation;

/**
 * @decs: 动画监听
 * @author: 郑少鹏
 * @date: 2019/9/24 11:45
 */
public interface AnimationListener {
    /**
     * We need to make MaterialIntroView visible before fade in animation starts.
     */
    interface OnAnimationStartListener {
        /**
         * Animation start.
         */
        void onAnimationStart();
    }

    /**
     * We need to make MaterialIntroView invisible after fade out animation ends.
     */
    interface OnAnimationEndListener {
        /**
         * Animation end.
         */
        void onAnimationEnd();
    }
}

File: library/src/main/java/com/zsp/library/jellytoolbar/listener/AnimationListener.kt
package com.zsp.library.jellytoolbar.listener

import android.animation.Animator

/**
 * @decs: AnimationListener
 * @author: 郑少鹏
 * @date: 2019/6/18 15:39
 */
abstract class AnimationListener : Animator.AnimatorListener {
    override fun onAnimationRepeat(animation: Animator?) = Unit
    override fun onAnimationStart(animation: Animator?) = Unit
    override fun onAnimationCancel(animation: Animator?) = Unit
    abstract override fun onAnimationEnd(animation: Animator?)
}